contrib/db_pools/lib/src/error.rs

```diff
@@ -0,0 +1,40 @@
+use std::borrow::Cow;
+use std::fmt;
+
+/// A general error type designed for the `Poolable` trait.
+///
+/// [`Pool::initialize`] can return an error for any of several reasons:
+///
+///   * Missing or incorrect configuration, including some syntax errors
+///   * An error connecting to the database.
+///
+/// [`Pool::initialize`]: crate::Pool::initialize
+#[derive(Debug)]
+pub enum Error<E> {
+    /// An error in the configuration
+    Config(Cow<'static, str>),
```

Comment:
This should be a `figment::Error`. And this should implement `From<figment:Error>` instead of the `Error::config()` method.